### PR TITLE
GEODE-7964: Upgrade Mockito to 3.3.3

### PR DESCRIPTION
**Pull request by kirklund**
GEODE-7964: Upgrade Mockito to 3.3.3

Fixup all tests touched for Mockito upgrade.
Update usage of Mockito deprecated APIs.

### CODE REVIEW

**Comment by kirklund** [comment]
Note: 2.23.4 and 2.24.0 passed IntegrationTest. 2.24.5 started to see OOME in integrationTest. First OOME was HARegionQueueIntegrationTest and then the *ConcurrentTests using Mockito.

HARegionQueueIntegrationTest was fixed by adding these lines to tearDown:
```
Mockito.framework().clearInlineMocks();
Mockito.reset();
```

And using stubOnly when creating mocks:
```
Mockito.mock(classToMock, withSettings().stubOnly());
```

**Comment by kirklund** [comment]
DistributedTest failure is JMXMBeanReconnectDUnitTest

**Comment by kirklund** [comment]
The Concurrent Tests that don't use Mockito still seem to hang in StressNewTest (I removed `@Ignore` so they now need to pass StressNewTest).

**Comment by kirklund** [comment]
Another unrelated JMXMBeanReconnectDUnitTest failure.

File: geode-core/src/main/java/org/apache/geode/cache/client/internal/ConnectionImpl.java [hunk 1]
```diff
@@ -223,6 +223,16 @@ public Socket getSocket() {
     return theSocket;
   }
 
+  @Override
+  public long getBirthDate() {
+    return 0; // TODO:KIRK
```
Comment (upthewaterspout) [inline geode-core/src/main/java/org/apache/geode/cache/client/internal/ConnectionImpl.java:228]: Clean up these TODO:KIRK parts.

Comment (kirklund) [reply geode-core/src/main/java/org/apache/geode/cache/client/internal/ConnectionImpl.java:228]: Thanks for catching those!

Comment (kirklund) [reply geode-core/src/main/java/org/apache/geode/cache/client/internal/ConnectionImpl.java]: Removed

**Pull request merged**